app/carbonapi/routes.go

```diff
@@ -4,57 +4,64 @@ import (
-	r.HandleFunc("/debug/pprof/", pprof.Index)
-	r.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
-	r.HandleFunc("/debug/pprof/profile", pprof.Profile)
-	r.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
-	r.HandleFunc("/debug/pprof/trace", pprof.Trace)
+	r.HandleFunc("/debug/pprof", pprof.Index)
+	s := r.PathPrefix("/debug/pprof").Subrouter()
+	s.HandleFunc("/cmdline", pprof.Cmdline)
+	s.HandleFunc("/profile", pprof.Profile)
+	s.HandleFunc("/symbol", pprof.Symbol)
+	s.HandleFunc("/trace", pprof.Trace)
 
 	r.Handle("/metrics", promhttp.Handler())
 
-	return r
+	return routeMiddleware(r)
 }
 
 func initHandlers(app *App) http.Handler {
-	r := http.NewServeMux()
+	r := mux.NewRouter()
 
-	r.HandleFunc("/render/", httputil.TimeHandler(app.validateRequest(http.HandlerFunc(app.renderHandler), "render"), app.bucketRequestTimes))
-	r.HandleFunc("/render", httputil.TimeHandler(app.validateRequest(http.HandlerFunc(app.renderHandler), "render"), app.bucketRequestTimes))
+	r.HandleFunc("/render", httputil.TimeHandler(
+		app.validateRequest(http.HandlerFunc(app.renderHandler), "render"), app.bucketRequestTimes))
 
-	r.HandleFunc("/metrics/find/", httputil.TimeHandler(app.validateRequest(http.HandlerFunc(app.findHandler), "find"), app.bucketRequestTimes))
-	r.HandleFunc("/metrics/find", httputil.TimeHandler(app.validateRequest(http.HandlerFunc(app.findHandler), "find"), app.bucketRequestTimes))
+	r.HandleFunc("/metrics/find", httputil.TimeHandler(
+		app.validateRequest(http.HandlerFunc(app.findHandler), "find"), app.bucketRequestTimes))
 
-	r.HandleFunc("/info/", httputil.TimeHandler(app.validateRequest(http.HandlerFunc(app.infoHandler), "info"), app.bucketRequestTimes))
-	r.HandleFunc("/info", httputil.TimeHandler(app.validateRequest(http.HandlerFunc(app.infoHandler), "info"), app.bucketRequestTimes))
+	r.HandleFunc("/info", httputil.TimeHandler(
+		app.validateRequest(http.HandlerFunc(app.infoHandler), "info"), app.bucketRequestTimes))
 
 	r.HandleFunc("/lb_check", httputil.TimeHandler(app.lbcheckHandler, app.bucketRequestTimes))
 
 	r.HandleFunc("/version", httputil.TimeHandler(app.versionHandler, app.bucketRequestTimes))
-	r.HandleFunc("/version/", httputil.TimeHandler(app.versionHandler, app.bucketRequestTimes))
 
 	r.HandleFunc("/functions", httputil.TimeHandler(app.functionsHandler, app.bucketRequestTimes))
-	r.HandleFunc("/functions/", httputil.TimeHandler(app.functionsHandler, app.bucketRequestTimes))
 
 	r.HandleFunc("/tags/autoComplete/tags", httputil.TimeHandler(app.tagsHandler, app.bucketRequestTimes))
 
 	r.HandleFunc("/", httputil.TimeHandler(app.usageHandler, app.bucketRequestTimes))
 
-	return r
+	return routeMiddleware(r)
 }
+
+// routeHelper formats the route using regex to accept optional trailing slash
+func routeMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		r.URL.Path = strings.TrimSuffix(r.URL.Path, "/")
+		next.ServeHTTP(w, r)
+	})
+}
```

Comment:
End file with newline.
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline